src/blocks/scribe-plugins/scribe-link-prompt-plugin.js

```diff
@@ -0,0 +1,161 @@
+"use strict";
+
+const scribeLinkPromptPlugin = function(block) {
+  // ===== INIT ===== //
+  var block = block || {};
+
+  if (!block.transforms) {
+    block.transforms = {};
+  }
+
+  ['pre', 'post'].forEach(function(key) {
+    if (!block.transforms[key]) {
+      block.transforms[key] = [];
+    }
+  });
+
+  // ===== PROMPTS ===== //
+  var userPrompts = [
+    {
+      // For emails we just look for a `@` symbol as it is easier.
+      regexp: /@/,
+      message: 'The URL you entered appears to be an email address. ' +
+      'Do you want to add the required “mailto:” prefix?',
+      action: function(link) {
+        return 'mailto:' + link;
+      }
+    },
+    {
+      // For tel numbers check for + and numerical values
+      regexp: /\+?\d+/,
+      message: 'The URL you entered appears to be a telephone number. ' +
+                'Do you want to add the required “tel:” prefix?',
+      action: function(link) {
+        return 'tel:' + link;
+      }
+    },
+    {
+      regexp: /.+/,
+      message: 'The URL you entered appears to be a link. ' +
+                'Do you want to add the required “http://” prefix?',
+      action: function(link) {
+        return 'http://' + link;
+      }
+    }
+  ];
+
+  function processPrompt(window, link) {
+    for (var i = 0; i < userPrompts.length; i++) {
+      var prompt = userPrompts[i];
+
+      if (prompt.regexp.test(link)) {
+        var userResponse = window.confirm(prompt.message);
+
+        if (userResponse) {
+          // Only process the first prompt
+          return prompt.action(link);
+        }
+      }
+
+    };
+
+    return link;
+  }
+
+  // ===== CHECKS ===== //
+  var urlProtocolRegExp = /^https?\:\/\//;
+  var mailtoProtocolRegExp = /^mailto\:/;
+  var telProtocolRegExp = /^tel\:/;
+  var knownProtocols = [urlProtocolRegExp, mailtoProtocolRegExp, telProtocolRegExp];
+
+  function emptyLink(string) {
+    return /\w/.test(string);
+  }
+
+  function hasKnownProtocol(urlValue) {
+    // If a http/s or mailto link is provided, then we will trust that an link is valid
+    return knownProtocols.some(function(protocol) { return protocol.test(urlValue)});
+  }
+
+  // ===== TRANSFORMS ===== //
+  function runTransforms(transforms, initialLink) {
+    return transforms.reduce(function(currentLinkValue, transform) {
+      return transform(currentLinkValue);
+      }, initialLink);
+  }
+
+  return function(scribe) {
+    const linkPromptCommand = new scribe.api.Command('link');
```

Comment:
I think it's better to keep the command as `linkPrompt`. If not the toolbar will need to be changed too and a migration path for old installs.